src/main/java/org/opentripplanner/updater/GraphUpdaterManager.java

```diff
@@ -218,16 +218,22 @@ private void reportReadinessForUpdaters() {
     Executors
       .newSingleThreadExecutor()
       .submit(() -> {
-        while (true) {
+        boolean otpIsShuttingDown = false;
+
+        while (!otpIsShuttingDown) {
           try {
             if (updaterList.stream().allMatch(GraphUpdater::isPrimed)) {
               LOG.info("OTP UPDATERS INITIALIZED - OTP is ready for routing!");
               return;
             }
             //noinspection BusyWait
             Thread.sleep(1000);
-          } catch (Exception e) {
+          } catch (RuntimeException e) {
```

Comment:
Do we really want to catch `RuntimeException`? Aren't those meant to stop the JVM?

Reply:
It's fine. `RuntimeException` is the base-class for all unchecked exceptions, for which lots of instances can occur during quite normal operations (like `NoSuchElementException` for example).

On this subject, while I agree that there are restrictions to its application, I don't think catching `Throwable` is automatically bad form. I tend to catch all errors on the top-level of the module or thread and log with as much pertinent information as possible from the state. If the alternative is to let the error propagate upwards to either the JVM or the execution framework (like Spring or an executor service) I rather log it myself to make sure the information is sent through the logging system in use. (Propagation of stdout/stderr is not always setup correctly.)

Catching `Throwable` and then continue execution is a no-no though, we'll agree on that.

Reply:
This is always safe to catch RuntimeException instead of Exception since compilation would fail if the block threw any checked exception. In this part of the code catching Exception is problematic since it could swallow InterruptedException and prevent graceful shutdown. And this is exactly what happened in this method. 

But I can also replace :

```

        catch (RuntimeException e) {
            LOG.error(e.getMessage(), e);
          } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            otpIsShuttingDown = true;
            LOG.info("OTP is shutting down, cancelling wait for updaters readiness.");
          }
```
          
by:          
          
```
         catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            otpIsShuttingDown = true;
            LOG.info("OTP is shutting down, cancelling wait for updaters readiness.");
          } catch (Exception e) {
            LOG.error(e.getMessage(), e);
          }
```
          
which is equivalent and maybe less confusing.
What do you think?

Reply:
@vpaturet I think the suggested second form is indeed clearer.